sql/core/src/main/scala/org/apache/spark/sql/Arrow.scala

```diff
@@ -41,7 +41,7 @@ object Arrow {
       case LongType => new ArrowType.Int(8 * LongType.defaultSize, true)
       case FloatType => new ArrowType.FloatingPoint(FloatingPointPrecision.SINGLE)
       case DoubleType => new ArrowType.FloatingPoint(FloatingPointPrecision.DOUBLE)
-      case ByteType => new ArrowType.Int(8, false)
+      case ByteType => new ArrowType.Int(8, true)
```

Comment:
Looks like scala treats bytes as signed int:

scala> scala.Byte.MinValue
res2: Byte = -128

scala> scala.Byte.MaxValue
res3: Byte = 127

So yes I think this is right